### PR TITLE
custom tree components

### PR DESCRIPTION
**Pull request by stefankoserkov**

### CODE REVIEW

File: src/tree/related-data/related-data-custom.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Component, Input } from "@angular/core";
```
Comment (Novkirishki) [inline src/tree/related-data/related-data-custom.component.ts:1]: Looks nice from usability point of view (clap)

**Review by annix8 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -228,6 +228,8 @@ Take a look at the following overview of the Admin App extension samples we prov
 
 * [Admin App custom theme](./src/theme#custom-theme-for-sitefinity-cms-admin-app) - You can customize the appearance of the Admin App by modifying specific components of the user interface. For example, you can customize buttons’ color, background, and text, as well as other supplementary text on the UI.
 
+* [Custom components loaded in the tree](./src/tree#custom-components-loaded-in-the-tree) - This extension is used to replace a part of each item shown in the tree used throughout the AdminApp. Currently supported only for the related data functionality.
+
```
Comment (Novkirishki) [inline README.md:232]: Users do not have a notion of tree and do not know where are we using it. Lets rename it to Custom list components

File: src/tree/README.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
+# Custom components loaded in the tree
+
+## Intro
+The tree component in the Sitefinity AdminApp is used in various places in the application, for example, the grid (the list of content items), the sidebar, related data and so forth. This extension will allow you to inject a custom Angualr component in the tree's item template. The picture below shows where the custom component will be rendered in the tree's item template.
```
Comment (Novkirishki) [inline src/tree/README.md:4]: Same here

File: src/tree/README.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+### Demo for related data
+The custom component for the related data feature will be rendered, in the following parts of the UI
+* already selected related items
+* dropdown selector for the recent items and adjacent search feature
+* related data selector and adjacent search feature
+* readonly mode of the content item
+
+![Demo](./../../assets/tree/RelatedDataCustomTreeComponent.gif)
+
+The custom component you develop will get the currently rendered content item data passed into it, for example, the item passed in to the sample custom component has this json representation:
```
Comment (Novkirishki) [inline src/tree/README.md:17]: aren't we passing a DataItem?

File: src/tree/README.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+* "hierarchy-taxa"
+* "images"
+* "listitems"
+* "lists"
+* "newsitems"
+* "pages"
+* "sites"
+* "taxonomies"
+* "videolibraries"
+* "videos"
```
Comment (Novkirishki) [inline src/tree/README.md:148]: Do we really need this list?

Comment (Novkirishki) [reply src/tree/README.md:148]: I believe they already know how to get the entity set name and these explanations are obsolete

File: src/tree/related-data/related-data-custom-tree-node-component-provider.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { ClassProvider } from "@angular/core";
+import { ComponentData } from "@progress/sitefinity-adminapp-sdk/app/api/v1/index-component/component-data";
+import { TreeNodeComponentProvider , CUSTOM_TREE_COMPONENT_TOKEN } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree";
+import { TreeNodeComponentFeatures } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree/custom-tree-node-component-features";
+import { RelatedDataCustomComponent } from "./related-data-custom.component";
+
+export class RelatedDataTreeNodeComponentProvider implements TreeNodeComponentProvider {
+    componentDataMap: Map<string, any>;
```
Comment (Novkirishki) [inline src/tree/related-data/related-data-custom-tree-node-component-provider.ts:8]: why do we need the maps. Can't we just return the desired component from the getComponentData() when the type is newsitems. That is the way we do it in other extensibility providers. Lets be consistent

File: src/tree/README.md (hunk 4 above)
Comment (stefankoserkov) [reply src/tree/README.md:17]: yes we are, removing this entire section

File: src/tree/README.md (hunk 3 above)
Comment (stefankoserkov) [reply src/tree/README.md:4]: done

File: README.md (hunk 2 above)
Comment (stefankoserkov) [reply README.md:232]: sure, i will change it

File: src/tree/README.md (hunk 5 above)
Comment (stefankoserkov) [reply src/tree/README.md:148]: removing

File: src/tree/related-data/related-data-custom-tree-node-component-provider.ts (hunk 6 above)
Comment (stefankoserkov) [reply src/tree/related-data/related-data-custom-tree-node-component-provider.ts:8]: sure

File: tests/e2e/app-elements/item-details.component.ts [hunk 7]
```diff
@@ -119,6 +119,17 @@ export class ItemDetails {
         expect(customFieldContent.trim()).toBe(fieldValue);
     }
 
+    static async VerifyCustomTreeNodeComponent(){
```
Comment (IrenaPankova) [inline tests/e2e/app-elements/item-details.component.ts:122]: Add Promise<void>

Comment (stefankoserkov) [reply tests/e2e/app-elements/item-details.component.ts:122]: added

File: src/tree/related-data/related-data-custom-tree-node-component-provider.ts (hunk 6 above)
Comment (stefankoserkov) [reply src/tree/related-data/related-data-custom-tree-node-component-provider.ts]: I simplified it

File: src/tree/README.md (hunk 5 above)
Comment (stefankoserkov) [reply src/tree/README.md]: removed

File: tests/e2e/app-elements/item-details.component.ts [hunk 8]
```diff
@@ -119,6 +119,17 @@ export class ItemDetails {
...
+        await BrowserWaitForElement(ItemDetailsMap.AddRelatedDataNewsItemsButton);
+        await ItemDetailsMap.AddRelatedDataNewsItemsButton.click();
+        await BrowserWaitForElement(ItemDetailsMap.FirstRelatedDataItem);
+        const createdOn = await ItemDetailsMap.FirstRelatedDataItem.findElement(by.css("span[data-sftest=\"custom-created-by\"]"));
```
Comment (IrenaPankova) [inline tests/e2e/app-elements/item-details.component.ts:126]: This is failing on the CI with findElement is not a function. Use element instead

File: README.md (hunk 2 above)
Comment (Novkirishki) [reply README.md:232]: Suggestion: This extensibility point is used to cistomize the visualization of items in the lists throughout the AdminApp. You can provide a custom component to replace the default list node template. Currently supported only for the related data functionality.

File: src/tree/related-data/related-data-custom-tree-node-component-provider.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { ClassProvider } from "@angular/core";
+import { ComponentData } from "@progress/sitefinity-adminapp-sdk/app/api/v1/index-component/component-data";
+import { TreeNodeComponentProvider , CUSTOM_TREE_COMPONENT_TOKEN } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree";
+import { TreeNodeComponentFeatures } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree/custom-tree-node-component-features";
+import { RelatedDataCustomComponent } from "./related-data-custom.component";
+
+export class RelatedDataTreeNodeComponentProvider implements TreeNodeComponentProvider {
+    getComponentData(feature: TreeNodeComponentFeatures, entitySet: string): ComponentData {
+        if (feature === TreeNodeComponentFeatures.RelatedData && entitySet === "newsitems") {
```
Comment (Novkirishki) [inline src/tree/related-data/related-data-custom-tree-node-component-provider.ts:9]: how simple it is - love it

File: tests/e2e/app-elements/item-details.component.ts (hunk 8 above)
Comment (stefankoserkov) [reply tests/e2e/app-elements/item-details.component.ts]: thank you, i changed it

File: src/tree/related-data/related-data-custom-tree-node-component-provider.ts (hunk 9 above)
Comment (stefankoserkov) [reply src/tree/related-data/related-data-custom-tree-node-component-provider.ts:9]: thank you

**Review by Novkirishki — APPROVED** [review]

**Pull request merged**